setup_and_run.sh

```diff
@@ -0,0 +1,238 @@
+#!/bin/bash
+# setup_and_run.sh
+# This script handles setting up the environment and running the Near AI Hub
+# It includes setup, database migrations, and running the application
+# Log function for better visibility
+log() {
+    echo "$(date '+%Y-%m-%d %H:%M:%S') - $1"
+}
+# Error handling
+set -e
+trap 'log "Error occurred at line $LINENO. Command: $BASH_COMMAND"' ERR
+# Step 1: Make sure we're in the correct directory
+check_directory() {
+    local path=$1
+
+    if [[ ! -d $path ]]; then
```

Comment:
If the directory doesn't exist, then we `cd` to it?  I think this is backwards from what you were trying to do.

Reply:
Yes, agree